framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -291,11 +359,22 @@ public boolean isSubtype(AnnotationMirror rhs, AnnotationMirror lhs) {
                 return false;
             } else if (AnnotationUtils.areSameIgnoringValues(lhs, rhs)) {
                 // Same type, so might be subtype
-                List<Object> lhsValues =
-                        AnnotationUtils.getElementValueArray(lhs, "value", Object.class, true);
-                List<Object> rhsValues =
-                        AnnotationUtils.getElementValueArray(rhs, "value", Object.class, true);
-                return lhsValues.containsAll(rhsValues);
+                if (AnnotationUtils.areSameByClass(rhs, IntRange.class)) {
+                    // Special case for IntRange
+                    Range lhsRange = getIntRange(lhs);
+                    Range rhsRange = getIntRange(rhs);
+                    if (lhsRange.from <= rhsRange.from && lhsRange.to >= rhsRange.to) {
```

Comment:
Simply write `return lhsRange.from <= rhsRange.from && lhsRange.to >= rhsRange.to` instead of the 5 lines for the conditional.